syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// define a service
service Greeter {
    // define the interface and data type
    rpc SayHello (HelloRequest) returns (HelloReply) {
    }

    // Sends another greeting
    rpc SayHelloAgain (HelloRequest) returns (HelloReply) {
    }
}

// define the data type of request
message HelloRequest {
    string name = 1;
}

// define the data type of response
message HelloReply {
    string message = 1;
}